### ISSUE TITLE
Does computing split gain account for the regularization of number of leaves? 

### ISSUE THREAD

**Issue by nd7141** [microsoft/LightGBM#5032]
## Description
Following [xgboost paper](https://arxiv.org/abs/1603.02754) (I assume LightGBM has the same logic) the loss reduction at split is given by:

![image](https://user-images.githubusercontent.com/4607709/155725599-76cd90df-ab96-4c66-8c1e-1652558fdf60.png)

I wonder if the gamma parameter at the end is accounted in LightGBM when computed split gain? If yes, what the value of this gamma? (because I could not find in parameters the values for this). 

The reason why this gamma could be important is that we could stop splitting earlier if gamma becomes more than the sum on the left. 

I also reckon that this gamma parameter is different from the number of leaves in a tree, because num_leaves is a hard-coded threshold which stops building a tree once the num leaves is saturated, while gamma can control the depth more adaptively (some trees will have many leaves, other fewer). 

Could you please clarify on the role of gamma in LightGBM?

**Comment by shiyu1994** [comment]
@nd7141 Thanks for using LightGBM. `gamma` is exactly `min_gain_to_split` in LightGBM. Note that the gain of every split condition will have `minus \gamma` according to the formula, so `\gamma` won't account for the selection of optimal split. It only influences when a split should be considered (I think in XGBoost paper, on split with positive `L_split` is considered). That's why `gamma` is `min_gain_to_split` in LightGBM.

**Comment by shiyu1994** [comment]
@nd7141 If you have any further questions, please feel free to post here, thanks.

**Comment by nd7141** [comment]
Thanks!
